proof/invariant-abstract/DetSchedSchedule_AI.thy

```diff
@@ -3149,15 +3152,21 @@ lemma map_fst_filter_zip_map_reduce:
   by (induct xs) auto
 
 lemma tcb_release_enqueue_wp':
-  "\<lbrace>\<lambda>s. tcb_sc_refill_cfgs_of s t \<noteq> None
-         \<longrightarrow> P (release_queue_update (tcb_release_enqueue_upd (tcb_ready_times_of s) t) s)\<rbrace>
+  "\<lbrace>\<lambda>s. tcb_sc_refill_cfgs_of s t \<noteq> None \<longrightarrow>
+          (\<forall>rt. tcb_ready_times_of s t = Some rt \<longrightarrow>
+            (if filter (\<lambda>(_, t'). t' \<le> rt) (zip (release_queue s) (map (the \<circ> tcb_ready_times_of s) (release_queue s))) = []
```

Comment:
I think it would be nice to have a definition (or maybe an abbreviation) for this filter, that the spec and proofs could both use. Depending on how it's written it might be able to be reused between both specs as well.
It's not a blocking issue, but if other people agree it could be worth a fixme/jira.

Reply:
Yeah, FIXME RT/JIRA sounds good